doc/code_snippets/test/msgpack/msgpack_object_iterator_take_test.lua

```diff
@@ -0,0 +1,18 @@
+local msgpack = require('msgpack')
+
+local mp_array = msgpack.object({ 10, 20, 30 })
+local mp_array_iterator = mp_array:iterator()
+
+local size = mp_array_iterator:decode_array_header()  -- returns 3
+local first = mp_array_iterator:decode()              -- returns 10
+mp_array_iterator:skip()                              -- returns none, skips 20
+local mp_array_new = mp_array_iterator:take()
```

Comment:
Why is it called `mp_array_new`? It's a MsgPack object that contains 30, i.e. it doesn't have anything to with MsgPack arrays.

Reply:
mixed up with `take_array`, thanks for pointing out